### ISSUE TITLE
Document CI platforms being used by Minishift

### ISSUE THREAD

**Issue by budhrg** [minishift/minishift#426]
See here https://github.com/minishift/minishift/issues/415#issuecomment-280667400

**Comment by thatdocslady** [comment]
@budhrg it looks like the comment you referenced discusses the CentOS CI doc which now exists (awesome!). Can you provide a bit more details about the other CI systems, not necessarily generic info but how we incorporate them in our specific build flow (I will include links to generic Travis/AppVeyor docs too).

**Comment by budhrg** [comment]
@thatdocslady http://bit.ly/minishiftci is the slides I prepare for one of my talk. Let me know if you need more help.

Slides definitely missing steps to setup any particular CI.

I can add steps to setup CentOS CI and for other CI @praveenkumar and @hferentschik can add those details.

**Comment by Preeticp** [comment]
The existing docs for CentOS CI are here: https://github.com/minishift/minishift/blob/master/docs/source/developing/ci.adoc
@hferentschik, @praveenkumar request you to add initial steps for other CI such as appveyor and circle CI as mentioned by @budhrg so that I can add it to the docs.

**Comment by praveenkumar** [comment]
@Preeticp For appveyor and CircleCI

```
You can find the appveyor CI job for Minishift here <https://ci.appveyor.com/project/hferentschik/minishift-o61ou>

The artifacts of a successful pull request build can be found at https://ci.appveyor.com/project/hferentschik/minishift-o61ou/build/<build_id>/aritfacts

Artifacts of a successful master build can be found at https://ci.appveyor.com/project/hferentschik/minishift-o61ou/branch/master/artifacts

```

```
You can find the circle-CI job for Minishift here <https://circleci.com/gh/minishift/minishift>

The artifacts of a successful pull request build can be found at https://circleci.com/gh/minishift/minishift/<build_id>#artifacts/  under below directory structure.
distiller/
    src/
        github.com/
            minishift/
                minishift/
                    out/

Artifacts of a successful master build can be found at https://circleci.com/gh/minishift/minishift/tree/master
```

**Comment by budhrg** [comment]
Also, @thatdocslady @Preeticp , also, it would be great if we mention which CI is used for testing which bits.

Like, CentOS CI is used specifically to test integration tests, unit tests, code format tests and cross build tests in Linux (CentOS). Other CI's AFAIR are:
- Circle CI : Used for testing unit tests, code format tests and cross build tests in macOS
- Appveyor CI: Used for testing unit tests, code format tests and cross build tests in Windows OS
- Travis CI : Used for testing unit tests, code format tests and cross build tests in Linux (Ubuntu)

**Comment by Preeticp** [comment]
Thanks @praveenkumar and @budhrg for the input.

**Comment by hferentschik** [comment]
Merged via pull request #1023
